fnl/tangerine/utils/path.fnl

```diff
@@ -5,15 +5,28 @@
 ; ALL() utils[env]
 (local env (require :tangerine.utils.env))
 (local p {})
+(local windows? (= _G.jit.os "Windows"))
+
+(lambda p.match [path pattern]
+  "matches 'path' against 'pattern' with support for windows."
+  (if windows?
+    (path:match (pattern:gsub "/" "[\\/]"))
+    (path:match pattern)))
+
+(lambda p.gsub [path pattern repl]
+  "substitutes 'pattern' in 'path' for 'repl' with support for windows."
+  (if windows?
+    (path:gsub (pattern:gsub "/" "[\\/]") repl)
+    (path:gsub pattern repl)))
```

Comment:
put this entire block of code under `Utils` header and not on the top level, top level is reserved for imports and module definition only.